phaseshifts/backends.py

```
@@ -0,0 +1,115 @@
+        if not slab_file:
+            raise BackendError("viperleed backend requires --slab <POSCAR>.")
+
+        try:
+            from viperleed.calc.files import parameters as viper_params
+            from viperleed.calc.files import phaseshifts as viper_phaseshifts
+            from viperleed.calc.files import poscar as viper_poscar
+            from viperleed.calc.psgen import runPhaseshiftGen
+        except ImportError:
+            raise BackendError(
+                "viperleed backend requires 'phaseshifts[viperleed]' to be installed."
+            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)